instr.ml

```diff
@@ -306,6 +308,17 @@ let lookup_fun (prog : program) (f : identifier) : afunction =
   try List.find (fun {name} -> name = f) prog.functions with
   | Not_found -> raise (FunctionDoesNotExist f)
 
+let update_fun ({main; functions} as prog : program) (func : afunction) : program =
```

Comment:
I did that because there is a function called `lookup_fun` and `update` seems to go along well with `lookup`. But its really a minor thing. Changed it to `replace_fun`.